### PR TITLE
Feat/update results

### PR DESCRIPTION
**Pull request by carmeltsadok**

### CODE REVIEW

File: AddEventResult/index.ts [hunk 1]
```diff
@@ -10,6 +10,20 @@ import authWrapper from '../shared/authWrapper';
 
 const functionName = 'addEventResult';
 
+// Union all notes object that been written by the same user (overrides the note content)
+const unionNotes = (notes: Note[]) => {
+    const unionedNotes: Note[] = [];
+    notes.forEach((noteObject: Note) => {
+        const searchedNote = unionedNotes.find(({ wroteBy }) => wroteBy.toString() === noteObject.wroteBy.toString());
+        if (searchedNote) {
+            searchedNote.note = noteObject.note;
```
Comment (PCOffline) [inline AddEventResult/index.ts:19]: why? You never used the searchedNote after that anyway

File: shared/questionnaireInstance/questionnaireInstance.manager.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * Unite Questionnaire schema with the answers. Throws errors if needed.
+ * @param requesterType - The requester type is Malshab or User
+ * @param answers - An array of answers to all questions. A question without an answer will get: "answer = undefined". The order of questions and answers matches.
+ * @param questionnaireSchema - The questions without answers.
+ * @param functionName - the name of function
+ * @returns
+ */
+// eslint-disable-next-line import/prefer-default-export
+export const solveQuestionnaire = (requesterType: RequesterTypes, answers: any[], questionnaireSchema: QuestionnaireSchema, functionName: string) => {
```
Comment (PCOffline) [inline shared/questionnaireInstance/questionnaireInstance.manager.ts:15]: Don't give `answers` the `any[]` type. Create an interface if needed, or use one if it already exists.

File: shared/questionnaireInstance/questionnaireInstance.manager.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    if (questionnaireSchema.questions.length !== answers.length) {
```
Comment (PCOffline) [inline shared/questionnaireInstance/questionnaireInstance.manager.ts:16]: What about questions that are not required?

**Review by PCOffline — CHANGES_REQUESTED** [review]

**Review by PCOffline — DISMISSED** [review]

**Review by Delitefully — APPROVED** [review]

File: GetQuestionnaireInstanceById/index.ts [hunk 4]
```diff
@@ -10,53 +10,49 @@ import NodeGroup from '../shared/nodeGroup/nodeGroup.interface';
...
+                if (!eventOfQuestionnaireInstance?.interviewersIds.includes(user.id.toString())) {
                     throw new ForbiddenFunctionError(functionName, user);
                 }
                 break;
             }
             case Role.Diagnoser:
+            case Role.Psychologist:
             case Role.RamadIturOfUnit: {
-                const eventOfQuestionnaireInstance: any = await EventModel.findOne({ 'results.questionnaire': req.params.id })
+                const eventOfQuestionnaireInstance: any = await EventModel.findOne({ 'results.questionnaire.questionnaireInstanceId': req.params.id })
```
Comment (PCOffline) [inline GetQuestionnaireInstanceById/index.ts:43]: Change this `any` to the appropriate interface

**Review by PCOffline — CHANGES_REQUESTED** [review]

**Review by Delitefully — APPROVED** [review]

**Review by PCOffline — APPROVED** [review]

**Pull request merged**